board/native: set TZ to UTC by default (#20023)
RIOT-OS/RIOT: boards/common/native/doc.md

- Network: Tap Interface
- UART: Runtime configurable - `/dev/tty*` are supported
- Timers: Host timer
- RTC: hosttime at 'UTC' to behave consitend with embedded systems if TZ is set that timzone will be used.
    - set `TZ` enviroment variable e.g.:
    - execute `export TZ=":/etc/localtime"` prior `make term` in the very shell you intend to run make term in
    - add `export TZ=":/etc/localtime"` to your `~/.profile`
    - run `TZ=":/etc/localtime" make term`

>     - set `TZ` in riot application picolibc and newlib are able to respect that setting if `+/-hh:mm:ss`

Reviewer: Can't we just add this to `pyterm` so it's less user hostile?

Author: i will not:  i think doing things wrong (make embedded things that assume no timezone behave wrong) should not be the default. < i think mktime behaving different on local and on embedded is much more user hostile

- Network: Tap Interface
- UART: Runtime configurable - `/dev/tty*` are supported
- Timers: Host timer
- RTC: host-time but at 'UTC' to behave consistent with embedded systems if TZ is set that timezone will be used.
    - set `TZ` environment variable e.g.:
    - execute `export TZ=":/etc/localtime"` prior `make term` in the very shell you intend to run make term in
    - add `export TZ=":/etc/localtime"` to your `~/.profile`
    - run `TZ=":/etc/localtime" make term`
    - to set set `TZ` in riot application as picolibc and newlib are able to respect that setting if `NAME+/-hh:mm:ss`
      (-time is added, +time is subtracted from UTC (+ pointing WEST England towards America) see `man timezone`

>       eg.: `IST-5:30` or `TZ="NZST-12NZDT-13,M10.1.0,M3.3.0"` or `TZ="CET-1CEST-2,M3.5.0,M10.5.0"` or `TZ="ACST-9:30ACDT-10:30,M10.1.0,M4.1.0"`

Reviewer 2: Thanks for the detailed documentation! It's a lot though now, and I'd say it warrants a separate subsection.

- Network: Tap Interface
- UART: Runtime configurable - `/dev/tty*` are supported
- Timers: Host timer

> - RTC: host-time but at `UTC` to behave consistent with embedded systems.

Reviewer 2: ```suggestion
- RTC: host time at `UTC` for consistent behavior with embedded systems.
```

>   If `TZ` environment is set that timezone will be used (see below for some ways to achieve that).

Reviewer 2: ```suggestion
  The environment variable `TZ` can be used to explicitly set a timezone (see [below](#setting-a-timezone)).
```

Author: ```suggestion
  The environment variable `TZ` can be used to explicitly set a specific or hosts timezone (see [below](#setting-a-timezone)).
```

As with any platform, you can specify the sizes of your stacks, i.e. the amount of space your application can use.
You may wish to use a more realistic stack size than native's `THREAD_STACKSIZE_DEFAULT` to increase realism.

> # Setting a timezone

Reviewer 2: ```suggestion
# Setting a timezone                                      {#setting-a-timezone}
```

and I've just learned doxygen doesn't do auto-anchors :/

RIOT native defaults to `UTC` timezone if `TZ` is not set in the environment,
to keep behavior of `time.h` functions like `mktime` consistent with embedded setups,
which usually do not use timezones.
If your application requires the same timezone as host, the `TZ` environment variable should be set accordingly.
There are several ways to achieve that:
- execute `export TZ=":/etc/localtime"` prior `make term`
  in the very shell you intend to run `make term` in
- add `export TZ=":/etc/localtime"` to your `~/.profile`
- run `TZ=":/etc/localtime" make term`

> The timezone can also be set at runtime by the application for both native and embedded devices using `setenv("TZ", <timezonestring>, 1)`. picolibc and newlibc expect `<timezonestring>` to have the form `NAME+/-hh:mm:ss<DST-handling>` (`+` pointing west -> `CET-1` adds 1 hour to UTC) (see `man timezone`)

Reviewer 2: ```suggestion
The timezone can also be set at runtime by the application for both native and embedded devices using `setenv("TZ", <timezonestring>, 1)`. `picolibc` and `newlibc` expect `<timezonestring>` to have the form `NAME+/-hh:mm:ss<DST-handling>` (`+` pointing west -> `CET-1` adds 1 hour to UTC, see `man timezone`)
```

> e.g.: "ACST-9:30ACDT-10:30,M10.1.0,M4.1.0"

Reviewer 2: ```suggestion
e.g., `ACST-9:30ACDT-10:30,M10.1.0,M4.1.0`.
```

> 
> all these allow you to make time function behave like host or another timezone.

Reviewer 2: ```suggestion
```

As with any platform, you can specify the sizes of your stacks, i.e. the amount of space your application can use.
You may wish to use a more realistic stack size than native's `THREAD_STACKSIZE_DEFAULT` to increase realism.

# Setting a timezone

> As mentioned above  RIOT native defaults to `UTC` timezone if `TZ` environment is not set.
> This is done to keep `time.h` functions like `mktime` to behave consistent with embedded setups,
> where `TZ` is usually not set.
> If your application requires the same timezone as host, the `TZ` environment variable should be set,
> there are numerous ways to achieve that:
> - execute `export TZ=":/etc/localtime"` prior `make term`
>   in the very shell you intend to run `make term` in
> - add `export TZ=":/etc/localtime"` to your `~/.profile`
> - run `TZ=":/etc/localtime" make term`

Reviewer 2: ```suggestion
RIOT native defaults to `UTC` timezone if `TZ` is not set in the environment,
to keep behavior of `time.h` functions like `mktime` consistent with embedded setups,
which usually do not use timezones.
If your application requires the same timezone as host, the `TZ` environment variable should be set accordingly.
There are several ways to achieve that:
- execute `export TZ=":/etc/localtime"` prior `make term`
  in the very shell you intend to run `make term` in
- add `export TZ=":/etc/localtime"` to your `~/.profile`
- run `TZ=":/etc/localtime" make term`
```

> `TZ` may also be set in application,
> it will then also be applied with newlib and picolibc `time.h`-functions on your embedded device.
> - use `setenv("TZ", <timezonestring>, 1)`to set `TZ` in riot application,
>   picolibc and newlib are able to respect that setting if `NAME+/-hh:mm:ss`
>   (-time is added, +time is subtracted from UTC (+ pointing WEST England towards America) see `man timezone`
>   eg.:
>       - `IST-5:30` or
>       - `TZ="NZST-12NZDT-13,M10.1.0,M3.3.0"` or
>       - `TZ="CET-1CEST-2,M3.5.0,M10.5.0"` or
>       - `TZ="ACST-9:30ACDT-10:30,M10.1.0,M4.1.0"`

Reviewer 2: ```suggestion
The timezone can also be set at runtime by the application for both native and embedded devices using `setenv("TZ", <timezonestring>, 1)`. picolibc and newlibc expect `<timezonestring>` to have the form `NAME+/-hh:mm:ss` (see `man timezone`).
```

(or do these examples have particular value over what people would get from `man timezone`?)

Author: Maybe we should at least provide one example either one common to most of the riot comunity (CET) or (feature packed one like ACST)
i would also like the + pointing west remark.

Author: ```suggestion
The timezone can also be set at runtime by the application for both native and embedded devices using `setenv("TZ", <timezonestring>, 1)`. picolibc and newlibc expect `<timezonestring>` to have the form `NAME+/-hh:mm:ss<DST-handling>` (`+` pointing west -> `CET-1` adds 1 horst to UTC) (see `man timezone`)
```

all these allow you to make time function behave like host or another timezone.
If the `TZ`-string can not be interpreted by the used `libc` (`newlib`,`picolibc`,`glibc`),
they default back to `UTC`.

> <span style="color:red">
> ! Some things will behave faulty, since the assumption of most pkgs and
> system modules is to run on a embedded system with no timezone set.
> One often used function that respect timezone setting is `mktime` !
> </span>

Reviewer 2: ```suggestion
@warning Some things will behave faulty, since the assumption of most pkgs and
        system modules is to run on a embedded system with no timezone set.
        One often used function that respect timezone setting is `mktime`
```

let's use doxygen styling instead
